src/lib/redux-cozy-api.js

```diff
@@ -0,0 +1,484 @@
+}
+
+const ids = (state = [], action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      if (!action.skip) {
+        return action.response.data.map(doc => doc.id)
+      }
+      return [
+        ...state,
+        ...action.response.data.map(doc => doc.id)
+      ]
+    case ADD_REFERENCED_FILES:
+      // TODO: here we don't follow JSONAPI conventions... (we pass an array of ids in the action)
+      return [
+        ...state,
+        ...action.ids
+      ]
+    case REMOVE_REFERENCED_FILES:
+      return state.filter(id => action.ids.indexOf(id) === -1)
+    case RECEIVE_CREATION_CONFIRM:
+      return [
+        ...state,
+        action.response.data[0].id
+      ]
+    case RECEIVE_DELETION_CONFIRM:
+      return state.filter(id => id !== action.response.data[0].id)
+    default:
+      return state
+  }
+}
+
+const endpoint = combineReducers({
+  type,
+  fetchStatus,
+  lastFetch,
+  hasMore,
+  count,
+  ids
+})
+
+const endpoints = (state = {}, action) => {
+  switch (action.type) {
+    case FETCH_DOCUMENTS:
+    case FETCH_REFERENCED_FILES:
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+    case ADD_REFERENCED_FILES:
+    case REMOVE_REFERENCED_FILES:
+    case RECEIVE_CREATION_CONFIRM:
+    case RECEIVE_DELETION_CONFIRM:
+    case RECEIVE_ERROR:
+      const entity = action.entity
+      const endpointKey = entity
+        ? (action.relationName ? `${entity.type}/${entity.id}/${action.relationName}` : `${entity.type}`)
+        : action.doctype
+      return Object.assign({}, state, { [endpointKey]: endpoint(state[endpointKey] || {}, action) })
+    default:
+      return state
+  }
+}
+
+export default combineReducers({
+  schemas,
+  entities,
+  endpoints
+})
+
+// utils
+const removeObjectProperty = (obj, prop) => {
+  return Object.keys(obj).reduce((result, key) => {
+    if (key !== prop) {
+      result[key] = obj[key]
+    }
+    return result
+  }, {})
+}
+
+const objectifyEntitiesArray = (entities) => {
+  let obj = {}
+  entities.forEach(entity => { obj[entity.id] = entity })
+  return obj
+}
+
+const mapEntitiesToIds = (entities, doctype, ids) => {
+  if (!entities[doctype]) {
+    return ids.map(() => null)
+  }
+  return ids.map(id => entities[doctype][id])
+}
+
+const slugify = (text) =>
+  text.toString().toLowerCase()
+    .replace(/\s+/g, '-')           // Replace spaces with -
+    .replace(/[^\w-]+/g, '')       // Remove all non-word chars
+    .replace(/--+/g, '-')         // Replace multiple - with single -
+    .replace(/^-+/, '')             // Trim - from start of text
+    .replace(/-+$/, '')             // Trim - from end of text
+
+const forceFileDownload = (href, filename) => {
+  const element = document.createElement('a')
+  element.setAttribute('href', href)
+  element.setAttribute('download', filename)
+  element.style.display = 'none'
+  document.body.appendChild(element)
+  element.click()
+  document.body.removeChild(element)
+}
```

Comment:
Because it is the only function that depends on a outside context. And that dependence pops up in my face with a little alert.